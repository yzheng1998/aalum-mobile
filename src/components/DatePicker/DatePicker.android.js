import React, { Component } from 'react'
import { DatePickerAndroid, View } from 'react-native'

export default class DatePicker extends Component {
  constructor(props) {
    super(props)
    this.state = {
      error: ''
    }
    const { setDate, doneOnPress, date } = props
    this.openDatePicker = async () => {
      try {
        const { action, year, month, day } = await DatePickerAndroid.open({
          date,
          mode: 'spinner'
        })
        if (action !== DatePickerAndroid.dismissedAction) {
          setDate(new Date(year, month, day))
          doneOnPress()
        } else if (action === DatePickerAndroid.dismissedAction) {
          doneOnPress()
        }
      } catch ({ code, message }) {
        this.setState({ error: `Cannot open date picker: ${message}` })
      }
    }
  }

  render() {
    return <View />
  }
}
