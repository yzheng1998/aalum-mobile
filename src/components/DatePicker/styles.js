import styled from 'styled-components'
import { themeGet } from 'styled-system'

export const DoneButton = styled.TouchableOpacity`
  height: 45px;
  width: 100%;
  justify-content: center;
  align-items: center;
  background-color: ${themeGet('colors.white')};
`

export const DoneButtonText = styled.Text`
  font-size: 16px;
  color: ${themeGet('colors.charcoal')};
  font-family: ${themeGet('fonts.bodyText')};
`

export const Container = styled.View`
  background-color: ${themeGet('colors.white')};
  position: absolute;
  align-self: center;
  bottom: 0;
  width: 100%;
`
