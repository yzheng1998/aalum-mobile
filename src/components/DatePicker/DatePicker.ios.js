import React, { Component } from 'react'
import { DatePickerIOS, View } from 'react-native'
import { DoneButton, DoneButtonText, Container } from './styles'

export default class DatePicker extends Component {
  openDatePicker = () => null

  render() {
    const { doneOnPress, date, setDate, mode, visible } = this.props
    return (
      <Container>
        {visible && (
          <View>
            <DoneButton onPress={doneOnPress}>
              <DoneButtonText>Done</DoneButtonText>
            </DoneButton>
            <DatePickerIOS mode={mode} date={date} onDateChange={setDate} />
          </View>
        )}
      </Container>
    )
  }
}
