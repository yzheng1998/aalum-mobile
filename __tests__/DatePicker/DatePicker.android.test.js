import React from 'react'
import DatePicker from '../../src/components/DatePicker/DatePicker.android'
import { mount } from 'enzyme'

describe('DatePicker Anrdoid', () => {
  test('renders correctly', () => {
    const wrapper = mount(<DatePicker />)
    expect(wrapper).toMatchSnapshot()

    const mockDate = new Date('01/01/1996')
    wrapper.setProps({ date: mockDate })
    expect(wrapper).toMatchSnapshot()
  })
})
