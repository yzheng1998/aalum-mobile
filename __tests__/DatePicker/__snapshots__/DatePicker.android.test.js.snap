// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Anrdoid renders correctly 1`] = `
<DatePicker>
  <View>
    <View />
  </View>
</DatePicker>
`;

exports[`DatePicker Anrdoid renders correctly 2`] = `
<DatePicker
  date={1996-01-01T05:00:00.000Z}
>
  <View>
    <View />
  </View>
</DatePicker>
`;
